huaweicloud/services/acceptance/secmaster/resource_huaweicloud_secmaster_incident_test.go

```diff
@@ -0,0 +1,198 @@
+package secmaster
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getIncidentResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getIncident: Query the SecMaster incident detail
+	var (
+		getIncidentHttpUrl = "v1/{project_id}/workspaces/{workspace_id}/soc/incidents/{id}"
+		getIncidentProduct = "secmaster"
+	)
+	getIncidentClient, err := cfg.NewServiceClient(getIncidentProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating SecMaster Client: %s", err)
+	}
+
+	getIncidentPath := getIncidentClient.Endpoint + getIncidentHttpUrl
+	getIncidentPath = strings.ReplaceAll(getIncidentPath, "{project_id}", getIncidentClient.ProjectID)
+	getIncidentPath = strings.ReplaceAll(getIncidentPath, "{workspace_id}",
+		fmt.Sprintf("%v", state.Primary.Attributes["workspace_id"]))
```

Comment:
the value type of  state.Primary.Attributes["workspace_id"] is string, it is better to use it directly

Reply:
sure